recipes/fontconfig/all/conanfile.py

```diff
@@ -75,8 +75,9 @@ def _patch_files(self):
     def build(self):
         # Patch files from dependencies
         self._patch_files()
-        autotools = self._configure_autotools()
-        autotools.make()
+        with tools.run_environment(self):
```

Comment:
Does it build by setting `self._autotools.libs = []` in `_configure_autotools`?

Reply:
I don't know for Linux, but on Macos, indeed it seems to be sufficient with libs = [](not need for tools.run_environment during configuration or build).
Though, on Macos with apple-clang 12, if all libs are shared, packaging of fontconfig fails:

```
 /usr/bin/install -c -m 644 source_subfolder/fonts.dtd '/Users/spaceim/.conan/data/fontconfig/2.13.92/_/_/package/38fe9038b485ccb26c394992e35513bce5519877/bin/share/xml/fontconfig'
dyld: Library not loaded: libfreetype.6.dylib
  Referenced from: /Users/spaceim/.conan/data/fontconfig/2.13.92/_/_/build/38fe9038b485ccb26c394992e35513bce5519877/fc-cache/.libs/fc-cache
  Reason: image not found
/bin/sh: line 1: 51798 Abort trap: 6           fc-cache/fc-cache -s -f -v
make[2]: *** [install-data-local] Error 134
make[1]: *** [install-am] Error 2
make: *** [install-recursive] Error 1

```

Reply:
Is this because of the generation of the font cache?
https://github.com/freedesktop/fontconfig/blob/ac3c9ba95db2657b20ea149011a1de7e435cb92c/Makefile.am#L140-L149

This also means that a native fontconfig package is different from a cross built one.

Reply:
@madebr In my setup (CentOS 8 with system `pkg-config` 1.4.2), clearing autotools library fixes the issue with clang not being able to load `libfreetyped.so.6` for a cross compilation check:
```
configure:3877: checking whether we are cross compiling
configure:3885: clang -o conftest -m64 -g -I/opt/conan/pkgconf/1.7.3/3rdparty/stable/package/fb49f7b339424395408c6d9629736e5998939ecf/include -I/opt/conan/pkgconf/1.7.3/3rdparty/stable/package/fb49f7b339424395408c6d9629736e5998939ecf/include/libpkgconf -I/opt/conan/freetype/2.10.4/3rdparty/stable/package/c7230691a40107e798ea6c568998b3ad8c169d99/include -I/opt/conan/freetype/2.10.4/3rdparty/stable/package/c7230691a40107e798ea6c568998b3ad8c169d99/include/freetype2 -I/opt/conan/expat/2.2.10/3rdparty/stable/package/fb49f7b339424395408c6d9629736e5998939ecf/include -I/opt/conan/libuuid/1.0.3/3rdparty/stable/package/b2b93ccb81f5d0dffc3804e641e0a022ead762c7/include -I/opt/conan/libuuid/1.0.3/3rdparty/stable/package/b2b93ccb81f5d0dffc3804e641e0a022ead762c7/include/uuid -I/opt/conan/libpng/1.6.37/3rdparty/stable/package/0fa8712e694ec5a8c92c3645e353174fe4bfb5cd/include -I/opt/conan/bzip2/1.0.8/3rdparty/stable/package/f1f068e94688b171cecdb59973de822d72b45612/include -I/opt/conan/brotli/1.0.9/3rdparty/stable/package/fb49f7b339424395408c6d9629736e5998939ecf/include -I/opt/conan/brotli/1.0.9/3rdparty/stable/package/fb49f7b339424395408c6d9629736e5998939ecf/include/brotli -I/opt/conan/zlib/1.2.11/3rdparty/stable/package/b2b93ccb81f5d0dffc3804e641e0a022ead762c7/include -m64 -L/opt/conan/pkgconf/1.7.3/3rdparty/stable/package/fb49f7b339424395408c6d9629736e5998939ecf/lib -L/opt/conan/freetype/2.10.4/3rdparty/stable/package/c7230691a40107e798ea6c568998b3ad8c169d99/lib -L/opt/conan/expat/2.2.10/3rdparty/stable/package/fb49f7b339424395408c6d9629736e5998939ecf/lib -L/opt/conan/libuuid/1.0.3/3rdparty/stable/package/b2b93ccb81f5d0dffc3804e641e0a022ead762c7/lib -L/opt/conan/libpng/1.6.37/3rdparty/stable/package/0fa8712e694ec5a8c92c3645e353174fe4bfb5cd/lib -L/opt/conan/bzip2/1.0.8/3rdparty/stable/package/f1f068e94688b171cecdb59973de822d72b45612/lib -L/opt/conan/brotli/1.0.9/3rdparty/stable/package/fb49f7b339424395408c6d9629736e5998939ecf/lib -L/opt/conan/zlib/1.2.11/3rdparty/stable/package/b2b93ccb81f5d0dffc3804e641e0a022ead762c7/lib conftest.c -lpkgconf -lfreetyped -lexpat -luuid -lpng16d -lm -lbz2 -lbrotlidec -lbrotlienc -lbrotlicommon -lz -lm >&5
configure:3889: $? = 0
configure:3896: ./conftest
./conftest: error while loading shared libraries: libfreetyped.so.6: cannot open shared object file: No such file or directory
configure:3900: $? = 127
configure:3907: error: in `/opt/conan/fontconfig/2.13.92/3rdparty/stable/build/4c058f1b8d81f641de297888dc8b6e16662bc79d':
configure:3909: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
```
But it still fails on checking Freetype's `.pc` file, seemingly due to requirement of a higher `pkg-config` version. The outcome of the discussion on Slack was that it is probably because the binary from the build_requirement (`pkgconf/1.7.3`) is not able to load the shared library from Conan, and is instead loading my system's `pkg-config` library, which has an older version:
```
configure:17051: checking for FREETYPE
configure:17119: result: no
configure:17121: error: in `/opt/conan/fontconfig/2.13.92/3rdparty/stable/build/4c058f1b8d81f641de297888dc8b6e16662bc79d':
configure:17123: error: The pkg-config script could not be found or is too old.  Make sure it
is in your PATH or set the PKG_CONFIG environment variable to the full
path to pkg-config.

Alternatively, you may set the environment variables FREETYPE_CFLAGS
and FREETYPE_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

To get pkg-config, see <http://pkg-config.freedesktop.org/>.
```

Reply:
Current master of fontconfig [requires a freeypte2 version >=21.0.15](https://github.com/freedesktop/fontconfig/blob/ac3c9ba95db2657b20ea149011a1de7e435cb92c/configure.ac#L317)

e.g. release freetype2 2.10.4 has [version 23:4:17 in its pc file](https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/builds/unix/configure.raw?h=VER-2-10-4#n20)

[Its meson.build is clear, it has 2 versions. A release version and a libtool version](https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/meson.build?h=VER-2-10-4#n31)

So an issue on this repo should be created that the pkg_config version of freetype2 is incorrect.

Reply:
what does the `config.log` file say?
Is there error that it cannot find the `fontconfig.pc`?
Or can't pkgconf run (because it cannot find `libpkgconf.so..x.y`?

Reply:
@madebr The snippet above is the failing part in `config.log`. I will post the complete logs requested in your other message. But as I said it makes sense that test program would not be able to run since linked libraries' paths are not in `LD_LIBRARY_PATH`. Do you not agree?